query/src/main/scala/filodb/query/exec/SetOperatorExec.scala

```diff
@@ -134,6 +128,13 @@ final case class SetOperatorExec(queryContext: QueryContext,
           if (resVal.key.labelValues == rv.key.labelValues) {
             rhsMap.put(jk, StitchRvsExec.stitch(rv, resVal, outputRvRange) )
           } else {
+            // TODO: Is it ok to overwrite the rhs of multiple RVs match the joinKey
+            //  Consider the case where two Range vectors match the join key and one of them has a non NaN value while
+            //  while other one has a NaN value for a given time step and the one with a non NaN value is the final RV
+            //  in the map. When we run this logic
+            //    val res = if (rhsRow.getDouble(1).isNaN) Double.NaN else lhsRow.getDouble(1) the rhs is not NaN and
```

Comment:
Looks like we shouldn't care whether-or-not the RHS value is `NaN`: http://demo.robustperception.io:9090/graph?g0.expr=vector(0)%20and%20vector(Nan)&g0.tab=1&g0.stacked=0&g0.range_input=1h

Reply:
Good catch but unfortunately we can't distinguish between ``vector(Nan)`` and value not being there which itself is ``NaN``